pilot/pkg/bootstrap/config_compare.go

```diff
@@ -27,11 +27,8 @@ import (
 // to be triggered. This is to avoid unnecessary pushes only when labels have changed
 // for example.
 func needsPush(prev config.Config, curr config.Config) bool {
-	if prev.GroupVersionKind != curr.GroupVersionKind {
```

Comment:
Why remove? May be fine if it 'never happens', but might be good to keep for safety. Or at least seems unrelated to cleaning constants

Reply:
Can not happen, so eliminate the compare